web3_proxy/src/bin/web3_proxy_cli/migrate_stats_to_v2.rs

```diff
@@ -0,0 +1,345 @@
+                            ),
+                            ..Default::default()
+                        }
+                    }
+                    None => AuthorizationChecks {
+                        ..Default::default()
+                    },
+                };
+
+                // Then overwrite rpc_key_id and user_id (?)
+                let authorization_type = AuthorizationType::Frontend;
+                let authorization = Arc::new(
+                    Authorization::try_new(
+                        authorization_checks,
+                        None,
+                        IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)),
+                        None,
+                        None,
+                        None,
+                        authorization_type,
+                    )
+                    .context("Initializing Authorization Struct was not successful")?,
+                );
+
+                // It will be like a fork basically (to simulate getting multiple single requests ...)
+                // Iterate through all frontend requests
+                // For each frontend request, create one object that will be emitted (make sure the timestamp is new)
+                let n = x.frontend_requests;
+
+                for i in 0..n {
+                    // info!("Creating a new frontend request");
+
+                    // Collect all requests here ...
+                    let mut int_request_bytes = (x.sum_request_bytes / n);
+                    if i == 0 {
+                        int_request_bytes += (x.sum_request_bytes % n);
+                    }
+
+                    let mut int_response_bytes = (x.sum_response_bytes / n);
+                    if i == 0 {
+                        int_response_bytes += (x.sum_response_bytes % n);
+                    }
+
+                    let mut int_response_millis = (x.sum_response_millis / n);
+                    if i == 0 {
+                        int_response_millis += (x.sum_response_millis % n);
+                    }
+
+                    let mut int_backend_requests = (x.backend_requests / n);
+                    if i == 0 {
+                        int_backend_requests += (x.backend_requests % n);
+                    }
+
+                    // Add module at the last step to include for any remained that we missed ... (?)
+
+                    // TODO: Create RequestMetadata
+                    let request_metadata = RequestMetadata {
+                        start_instant: Instant::now(),           // This is overwritten later on
+                        request_bytes: int_request_bytes.into(), // Get the mean of all the request bytes
+                        archive_request: x.archive_request.into(),
+                        backend_requests: Default::default(), // This is not used, instead we modify the field later
+                        no_servers: 0.into(), // This is not relevant in the new version
```

Comment:
shouldn't the new version track this, still? I thought it was in both tables.

Reply:
so we still do. This is a workaround because the constructor does not really accept these (calculates them according to some parameters).

That's why I introduced the `modify_struct` function to keep the constructor the same (because we will not use this outside the migration), where these values are overwritten by what is currently inserted in the database

Reply:
I don't see no_servers (or anything similar) in any of the tables. Please let me know if this is still relevant, will resolve it otherwise